source/manual/how-to-remove-jquery-from-javascript.html.md

```diff
@@ -0,0 +1,202 @@
+
+```javascript
+var jqExample = $('div')
+jqExample.addClass('test')
+```
+
+Using JS to find an element returns a node, which can have JS functions applied to it e.g.
+
+```javascript
+var jsExample = document.querySelector('div')
+jsExample.addClass('test') // errors
+jsExample.classList.add('test') // works
+```
+
+`jqExample` and `jsExample` are incompatible. jQuery functions cannot be applied to the JS node and vice versa.
+
+The jQuery object contains a JS node, accessed as `jqExample[0]`. Note that `jqExample` is not an iterable array.
+
+### Things where jQuery and JS use the same function name
+
+Some jQuery functions are named the same as their equivalent JS function, for example:
+
+- `click()`
+- `focus()`
+- `closest()`
+
+This is confusing, because the code chooses the right one to use based on what kind of object it’s being run on.
+
+```javascript
+$('button').click() // calls the jQuery function
+document.querySelector('button').click() // calls the JS
+```
+
+Note that some of these may behave slightly differently, particularly in what they return.
+
+### jQuery is more forgiving than JS
+
+The result of jQuery’s find function might not find anything, but it will still return an empty jQuery object. Subsequent actions on that object (e.g. adding a class) might not do anything but they won’t cause an error.
+
+Attempting the same thing with JS and `querySelector` will return `null` if no matching element is found, which means any attempt to do anything with that result will cause an error.
+
+This means more checks may have to be added to the code.
+
+```javascript
+var example = document.querySelector('.this')
+
+if (example) {
+  // do something
+}
+```
+
+### When to use querySelector and querySelectorAll
+
+In jQuery an element can be found on the page using `.find()`. There are two equivalent functions in JS.
+
+- [querySelector](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) finds only the first element that matches the given query.
+- [querySelectorAll](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) finds all elements that match that query, and returns an array of them.
+
+The code should suggest which should be used:
+
+- if the result of `find` is used in a loop, or an each, use `querySelectorAll`
+- if `find` is used for an id e.g. `.find('#wrapper')`, `use querySelector`
+- otherwise, look for clues - count the instances of elements that would match in the markup, is the variable name plural or singular?
+
+### Setting and getting attributes
+
+jQuery uses the same functions to both get and set attributes on elements.
+
+```javascript
+.attr('class') // replace with getAttribute('class')
+.attr('class', 'value') // replace with setAttribute('class', 'value')
+
+.data('item') // replace with getAttribute('data-item')
+.data('item', 'value') // replace with setAttribute('data-item', 'value')
+```
```

Comment:
Can the JavaScript equivalents be put in their own code block? This would follow the pattern set in the examples in "Chaining functions" and "jQuery objects are different from JS nodes".

Reply:
As a reference guide I thought it was clearer to keep them together, so readers can easily see the jQuery and JS equivalent. Happy to change to something else though if you can come up with an alternative - I can't think of a better way to do this.

Reply:
@injms I've had a rethink on this and tried your suggestion - please take a look.

Reply:
That's super and exactly what I had in mind - thank you